File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.java
class Solution {
    public int maxProfit(int[] prices) {
        int start = 0;
        int end = 1;
        int max = 0;
        while(end < prices.length) {
            if(prices[start] < prices[end]) {
                int difference = prices[end]-prices[start];
                max = Math.max(difference, max);
                end++;
            } else {
                start = end;
                end++;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-4.py
import math
class Solution:
    def maxProduct(self, nums: List[int]) -> int:

        res = float('-inf')
        
        for i in range(len(nums)):
            for j in range(i, len(nums)):
                subarray = nums[i:j+1]
                res = max(res, math.prod(subarray))
                
        return res